src/SayMore/MediaUtils/Audio/WavePainting/WavePainterBasic.cs

```diff
@@ -494,6 +494,12 @@ protected virtual void DrawWave(PaintEventArgs e, Rectangle rc)
 				{
 					var sampleAmplitudes = _samplesToDraw[sampleToDraw, channel];
 
+					if (sampleAmplitudes == null)
+					{
+						Logger.WriteEvent("'sampleAmplitudes' was null; 'sampleToDraw' or 'channel' possibly off by one. (See SP-1799)");
```

Comment:
Let's go ahead and include the values of the two indexes and the dimensions of the array also so we can see whether we're at a boundary case.

Reply:
Done.